src/xbase/layout/mos/data.py

```diff
@@ -232,8 +259,23 @@ def top_conn_types(self) -> Sequence[MOSWireType]:
         if self.flip:
             return MOSWireType.G, MOSWireType.G_MATCH
         else:
+            if self.double_gate:
+                return MOSWireType.G2, MOSWireType.G2_MATCH
+            return MOSWireType.DS_GATE, MOSWireType.DS, MOSWireType.DS_MATCH
+
+    @property
+    def mid_conn_types(self) -> Sequence[MOSWireType]:
+        """Return sequence of top wire connection types.
+
+        index 0 is the default type.
+        """
+        if self.double_gate:
             return MOSWireType.DS_GATE, MOSWireType.DS, MOSWireType.DS_MATCH
 
+        raise RuntimeError("trying to use mid conn, when its not a double gate")
+        return []
```

Comment:
The return statement is never reached because of the RunTimeError. PyCharm should have shown a warning here. Delete the return statement.